lib/versionmanager.js

```diff
@@ -456,7 +457,7 @@ function queryVersions(packageMap, options = {}) {
         });
     }
 
-    return Promise.all(packageList.map(getPackageVersionProtected))
+    return pMap(packageList,getPackageVersionProtected,{concurrency: os.cpus().length})
```

Comment:
I believe the CPU is not the bottleneck on HTTP requests.

> A key feature of node compared to some other systems is that the cost of holding open an idle connection is very low.  I've written tests that open up 64K connections, and once the connections are established, the CPU is obvious idle while no work is taking place, but the resident memory is under 200MB.

https://www.quora.com/How-many-connections-can-a-node-js-server-handle

Instead, I would suggest providing a reasonable default of `10`, and allow that to be overridden by a cli option.

Reply:
I searched the internet But I didn't find any good recommendation on how to decide the requests concurrency, the only thing I found is this old post:
https://www.physicsforums.com/threads/idm-how-to-increase-downloading-thread-number.875706/post-5501364

Which says that it should be at most 8(multi-threaded download managers almost have the same problem as ours).

I'm thinking of setting it to 8 without providing a cli option, I don't think any user is interested in changing this number, in fact I didn't know ncu issues requests in parallel until I looked at code, before that I thought it issues one request at a time.

Reply:
Yes, `8` should be fine. We are fetching from a known source, the `npm` registry, so it is less of a mystery how many connections they support (many), but I still think it's a good idea to batch requests.

Reply:
I do ask that this be added as a cli option. I am not willing to hard-code it into the library.

Reply:
finished, I made 8 the default and added cli option called `--requests-batch-size`, feel free to change its name and description.